device/simulatedtopology/simulatedtopology.yaml

```diff
@@ -0,0 +1,127 @@
+          type: string
+          enum:
+          - connected_to_st_router
+          - connected_to_em_router
+          - connected_to_external_router
+          default: connected_to_st_router
+        st_router_name: 
+          description: >-
+            The unique name of the simulated router to which this link is conected. 
+            Note that it can only be connected to another node of the same topology.  
+          type: string
+          x-constraint:
+            - '/components/schemas/Router.SimulatedTopologyNode/properties/name'
+        external_router_name:
+          description: >-
+            The unique name of the simulated router or emulated router in another topology to
+             which this link is conected. 
+          type: string
+          x-constraint:
+            - '/components/schemas/Router.SimulatedTopologyNode/properties/name'
+            - '/components/schemas/Router/name'
+        enable_ipv4_address:
+          description: >-
+            If this is enabled then endpoints of this link will have the Ipv4 To and From addresses.
+          type: boolean
+          default: true        
+        ipv4_from_address:
+            description: >-
+              The IPv4 From address.
+            type: string
+            format: ipv4
+            default: 0.0.0.0
+        ipv4_to_address:
+          description: >-
+            The IPv6 To address.
+          type: string
+          format: ipv4
+          default: 0.0.0.0
+        ipv4_prefix_lenght:
+          description: >-
+            Ipv4 prefix lenght of the From and To address
+          type: integer
+          minimum: 1
+          maximum: 32
+          default: 16
+        enable_ipv6_address:
+          description: >-
+            If this is enabled then endpoints of this link will have the Ipv6 To and From addresses.
+          type: boolean
+          default: false
+        ipv6_from_address:
+          description: >-
+            The IPv6 From address.
+          type: string
+          format: ipv6
+          default: ::0
+        ipv6_to_address:
+          description: >-
+            The IPv6 To address.
+          type: string
+          format: ipv6
+          default: ::0
+        ipv6_prefix_length:
+          description: >-
+            Ipv6 prefix lenght of the From and To address.
+          type: integer
+          minimum: 1
+          maximum: 128
+          default: 64
+        isis_sim_link:
+          description: >-
+              This contains the properties of a simulated Isis Link.
+          $ref: './isissimulatedtopology/isissimulatedtopologylink.yaml#/components/schemas/Router.IsisSimulatedTopology.Link'  
+        ospf_sim_link:
+          description: >-
+            This contains the properties of a simulated Ospf Link.
+          $ref: './ospfsimulatedtopology/ospfsimulatedtopologylink.yaml#/components/schemas/Router.OspfSimulatedTopology.Link'
```

Comment:
how are these simulated topology is mapped to the router instance?
is it okay to map these under isis and ospf routerange nodes? just like network topology under network group?

Reply:
Each Router has an array of Simulated Topologies.  Each Router has one ISIS router or one OSPF Router or both.
So the nodes in the ST(s) need to be advertised by either ISIS or OSPF or both. Note that OSPFv3 and LDP can also have properties to advertise in the ST per node/ or per node/link in the future. So this is modelled as a shared object that can be advertised by multiple protocols. It might be very cumbersome to duplicate the same ST twice for OSPF , ISIS and maybe other protocols as well.